### PR TITLE
Feature/sorting

### PR DESCRIPTION
**Pull request by AmeerAmjed**
implement change sort countries and tourist places **A to Z** and **Z to A**  with add DropDownMenu to change sort in screen.

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt [hunk 1]
```diff
@@ -5,20 +5,28 @@ import kotlinx.coroutines.CoroutineScope
...
+                val countries = getCountriesSorted(
+                    countriesApi.getCountriesList(),
+                    sortOrder
+                )
                 state.emit(
                     ListScreenState.Success(
-                        countriesList = countriesList,
-                        selectedCountry = countriesList.first(),
+                        countriesList = countries,
+                        selectedCountry = countries.first()
```
Comment (SEAbdulbasit) [inline shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt:29]: What about doing the sorting there directly like
```

val sortedCountriesList = if (isAscending) {
    countriesApi.getCountriesList().sortedBy { it.name }
} else {
    countriesApi.getCountriesList().sortedByDescending { it.name }
}
```

File: shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt [hunk 2]
```diff
@@ -73,6 +98,20 @@ class ListScreenViewModel {
...
+
+    override fun sortByNameDesc() {
+        fetchCountries(sortOrder = SortOrder.DESCENDING)
+    }
+}
+
+enum class SortOrder {
+    ASCENDING,
+    DESCENDING
 }
```
Comment (SEAbdulbasit) [inline shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt]: No need for interface i believe. We will send action from View to View Model
```

sealed interface Action{
      object SortAcending: Action
      object SortDecending:Action
}
```

Something like that. We will get action from view and we will handle that in view model. wdyt ?

**Review by SEAbdulbasit — CHANGES_REQUESTED** [review]

**Comment by SEAbdulbasit** [comment]
Also, it would be nice if you can add some screenshots or a short video.

File: shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt (hunk 1 above)
Comment (AmeerAmjed) [reply shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt:29]: @SEAbdulbasit I need to change sort by **Countries name** and **TouristPlaces name** in one function. My code handles this case.

File: shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt [hunk 3]
```diff
@@ -27,7 +35,23 @@ class ListScreenViewModel {
...
+    ): List<Country> {
+        return if (sortOrder == SortOrder.ASCENDING) {
+            countries.sortedBy { it.name }
+        } else {
+            countries.sortedByDescending { it.name }
+        }.map { country ->
+            country.copy(
+                touristPlaces = country.touristPlaces.sortedBy { it.name }
+            )
+        }
```
Comment (SEAbdulbasit) [inline shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt:51]: why map ?

Comment (SEAbdulbasit) [reply shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt:51]: There is no need for this. sortBy already give us a list as a result. We just need to return that.

**Review by SEAbdulbasit — CHANGES_REQUESTED** [review]

File: shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt (hunk 3 above)
Comment (AmeerAmjed) [reply shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt:51]: This map to sort tourist places by name from A to Z. @SEAbdulbasit

**Review by SEAbdulbasit — APPROVED** [review]

**Pull request merged**